src/EnergyPlus/Plant/Solution/PlantLoopSolver.cc

```diff
@@ -168,215 +158,248 @@ namespace PlantLoopSolver {
+        if ( CurrentLoopSideIsConstantSpeedBranchPumped ) {
+            // turn off any pumps connected to unloaded equipment and re-do the flow/load solution pass
+			thisPlantLoop.loopSolver.DisableAnyBranchPumpsConnectedToUnloadedEquipment(LoopNum, ThisSide);
+			thisPlantLoop.loopSolver.DoFlowAndLoadSolutionPass(LoopNum, ThisSide, OtherSide, ThisSideInletNode, FirstHVACIteration);
+        }
 
 		// A couple things are specific to which LoopSide we are on
-		if ( LoopSideNum == DemandSide ) {
+		if ( LoopSideNum == DataPlant::DemandSide ) {
 
 			// Pass the loop information via the HVAC interface manager
-			UpdatePlantLoopInterface( LoopNum, LoopSideNum, PlantLoop( LoopNum ).LoopSide( DemandSide ).NodeNumOut, PlantLoop( LoopNum ).LoopSide( SupplySide ).NodeNumIn, ReSimOtherSideNeeded, PlantLoop( LoopNum ).CommonPipeType );
+			HVACInterfaceManager::UpdatePlantLoopInterface( LoopNum, LoopSideNum, thisPlantLoop.LoopSide( DataPlant::DemandSide ).NodeNumOut, thisPlantLoop.LoopSide( DataPlant::SupplySide ).NodeNumIn, ReSimOtherSideNeeded, thisPlantLoop.CommonPipeType );
 
 		} else { //LoopSide == SupplySide
 
 			// Update pressure drop reporting, calculate total loop pressure drop for use elsewhere
-			SimPressureDropSystem( LoopNum, FirstHVACIteration, PressureCall_Update );
+			PlantPressureSystem::SimPressureDropSystem( LoopNum, FirstHVACIteration, DataPlant::PressureCall_Update );
 
 			// Pass the loop information via the HVAC interface manager (only the flow)
-			UpdatePlantLoopInterface( LoopNum, LoopSideNum, PlantLoop( LoopNum ).LoopSide( SupplySide ).NodeNumOut, PlantLoop( LoopNum ).LoopSide( DemandSide ).NodeNumIn, ReSimOtherSideNeeded, PlantLoop( LoopNum ).CommonPipeType );
+			HVACInterfaceManager::UpdatePlantLoopInterface( LoopNum, LoopSideNum, thisPlantLoop.LoopSide( DataPlant::SupplySide ).NodeNumOut, thisPlantLoop.LoopSide( DataPlant::DemandSide ).NodeNumIn, ReSimOtherSideNeeded, thisPlantLoop.CommonPipeType );
 
 			// Update the loop outlet node conditions
-			CheckLoopExitNode( LoopNum, FirstHVACIteration );
+			thisPlantLoop.loopSolver.CheckLoopExitNode( LoopNum, FirstHVACIteration );
 
 		}
 
 		// Update some reporting information at Plant half loop level
-		UpdateLoopSideReportVars( LoopNum, LoopSideNum, InitialDemandToLoopSetPointSAVED, LoadToLoopSetPointThatWasntMet );
+		thisPlantLoop.loopSolver.UpdateLoopSideReportVars( LoopNum, LoopSideNum, InitialDemandToLoopSetPointSAVED, LoadToLoopSetPointThatWasntMet );
 
 	}
 
-	//==================================================================!
-	//==================================================================!
-	//==================================================================!
-
-	//==================================================================!
-	//================= TOPOLOGY VALIDATION ROUTINE ====================!
-	//==================================================================!
+	void
```

Comment:
This new helper function re-enables all branches on the loop in case any were disabled as a part of this branch pumping fix.